header-footer-grid/Core/Components/Abstract_Component.php

```diff
@@ -148,6 +148,17 @@ public function __construct( $panel ) {
 		add_action( 'init', [ $this, 'define_settings' ] );
 	}
 
+	/**
+	 * Method to filter component loading if needed.
+	 *
+	 * @since   1.0.1
+	 * @access public
+	 * @return bool
+	 */
+	public function should_component_be_active() {
```

Comment:
you can use a different name, like `maybe_activate` or `is_active` without `component` in the method name as this is used in the class name. 
E.g: `$component->should_component_be_active` is using twice the `component`  word